cli/azd/internal/vsrpc/environment_service_create.go

```diff
@@ -67,7 +67,7 @@ func (s *environmentService) CreateEnvironmentAsync(
 			return false, fmt.Errorf("reading app host manifest: %w", err)
 		}
 
-		projectName := strings.TrimSuffix(filepath.Base(c.azdContext.ProjectDirectory()), ".AppHost")
+		projectName := strings.TrimSuffix(c.azdContext.GetDefaultProjectName(), ".AppHost")
```

Comment:
Just wondering, would this code still make sense the refactor to `GetDefaultProjectName`? `GetDefaultProjectName` is now an implementation detail that we then use to trim the`.AppHost` suffix?

Reply:
The current change actually regresses functionality -- we would have replaced `.` with `-` and `TrimSuffix("my-app-host", ".AppHost")` would not trim the actual suffix.

Reply:
My coded up suggestions here:

https://github.com/vhvb1989/azure-dev/compare/fix-projects-with-empty-spaces-aspire...weikanglim:weilim/aspire-naming?expand=1